From 27d3ce1467990f89126e228559dec8f84b96c60e Mon Sep 17 00:00:00 2001
From: "H.J. Lu" <hjl.tools@gmail.com>
Date: Fri, 1 Apr 2016 15:08:48 -0700
Subject: [PATCH] Remove Fast_Copy_Backward from Intel Core processors

Intel Core i3, i5 and i7 processors have fast unaligned copy and
copy backward is ignored.  Remove Fast_Copy_Backward from Intel Core
processors to avoid confusion.

	* sysdeps/x86/cpu-features.c (init_cpu_features): Don't set
	bit_arch_Fast_Copy_Backward for Intel Core proessors.
---
 ChangeLog                  | 5 +++++
 sysdeps/x86/cpu-features.c | 6 +-----
 2 files changed, 6 insertions(+), 5 deletions(-)

diff -rup a/sysdeps/x86/cpu-features.c b/sysdeps/x86/cpu-features.c
--- a/sysdeps/x86/cpu-features.c	2020-01-21 16:44:28.637555853 -0500
+++ b/sysdeps/x86/cpu-features.c	2020-01-21 16:46:51.208756416 -0500
@@ -116,11 +116,8 @@ init_cpu_features (struct cpu_features *
 	    case 0x2c:
 	    case 0x2e:
 	    case 0x2f:
-	      /* Rep string instructions, copy backward, unaligned loads
+	      /* Rep string instructions, unaligned load, unaligned copy,
 		 and pminub are fast on Intel Core i3, i5 and i7.  */
-#if index_Fast_Rep_String != index_Fast_Copy_Backward
-# error index_Fast_Rep_String != index_Fast_Copy_Backward
-#endif
 #if index_Fast_Rep_String != index_Fast_Unaligned_Load
 # error index_Fast_Rep_String != index_Fast_Unaligned_Load
 #endif
@@ -129,7 +126,6 @@ init_cpu_features (struct cpu_features *
 #endif
 	      cpu_features->feature[index_Fast_Rep_String]
 		|= (bit_Fast_Rep_String
-		    | bit_Fast_Copy_Backward
 		    | bit_Fast_Unaligned_Load
 		    | bit_Prefer_PMINUB_for_stringop);
 	      break;

